KarhooUISDK/Screens/CheckoutScreen/Components/Loyalty/LoyaltyViewModel.swift

```diff
@@ -89,7 +89,7 @@ class LoyaltyViewModel: ObservableObject {
             self.tripAmount = model.tripAmount
             self.burnAmount = model.burnAmount
             self.canBurn = model.canBurn
-            if model.canEarn {
+            if model.canEarn && worker.modeSubject.value == .none {
                 worker.modeSubject.send(.earn)
```

Comment:
does this mean that even when the user cannot earn, the sdk will act as if it could? maybe I'm being too cautious, but this sounds like a good place for some bugs to creep in. has this been tested in all loyalty combinations?

Reply:
Check removed due to behavior duplication